pylds/testing_scripts.py

```diff
@@ -68,3 +68,66 @@ def generate_pes_data(coords, potential, filename):
     hf.create_dataset('coords', data=np.array(coords).astype('float64'))
     hf.create_dataset('pes_data', data=pes_data.astype('float64'))
     hf.close()
+
+def discretise_vector_field(sample_time_points, sample_coords, vector_field):
+    """
+    Returns a 1-dimensional array of 2-D vector field values evaluated on a 2D grid of points 
+    for a 1-dimensional array of time-points. 
+
+    Parameters
+    ----------
+    sample_time_points: 1d array,
+        time-points in a sample time-interval.
+    
+    sample_coords : list of ndarrays,
+        [x,y] contain coordinates.
+
+    vector_field : function,
+        Function/vector field to be discretised.
+
+    Returns
+    -------
+    vector_field_data : ndarray, len(t),
+        Array of function/vector field values.
+    """
+    x, y = sample_coords
+    X,Y = np.meshgrid(x,y,indexing='xy')
+    sample_xy_points = np.column_stack([X.ravel(),Y.ravel()])
+    
+    vector_field_data = [vector_field(t, sample_xy_points) for t in sample_time_points]
+
+    return np.array(vector_field_data)
+
+def generate_vector_field_data(sample_time_points, sample_coords, vector_field, filename):
+    """
+    Saves 2-dimensional array of function (vector field) values on a grid of points to 
+    pylds/vector_field_files/filename.hdf5. File format fixed to HDF5 by default.
+
+    Note that storage fails if x_axis and y_axis in sample_coords are not the same length.
+
+    Parameters
+    ----------
+    sample_time_points: 1d array,
+        time-points in a sample time-interval.
+    
+    sample_coords : list of ndarrays,
+        [x,y] contain coordinates.
+
+    vector_field : function,
+        Function/vector field to be discretised.
+
+    filename : string
+        Name of hdf5 file to be saved.
+    """
+
+    vector_field_data = discretise_vector_field(sample_time_points, sample_coords, vector_field)
+    
+    dirname = 'pylds/vector_field_files'
+    filepath = os.path.join(dirname, filename+'.hdf5')
+    hf = h5py.File(filepath,'w')
+    
+    hf.create_dataset('sample_time', data=np.array(sample_time_points).astype('float64'))
+    hf.create_dataset('sample_coords', data=np.array(sample_coords).astype('float64'))
+    hf.create_dataset('vector_field_data', data=vector_field_data.astype('float64'))
```

Comment:
Depending on the magnitude of the expected interpolation error, it may be worth saving data as 'float32'.

Reply:
Yeah sure. However, would you do the same for the PES data too?

Reply:
Yes, in case it's worth doing so for the vector field, then it's also worth for PES data. I can run a quick test for PES data to see the difference it makes.

Reply:
Great!

Reply:
For the discretised Hénon-Heiles, float32 seems to be marginally slower in both fitting the spline and integrating trajectories. Let's disregard this suggestion of mine.

Reply:
OK, noted then.